File: book/JPA/chap08/src/main/java/com/example/chap08/entity/BaseEntity.java
package com.example.chap08.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.MappedSuperclass;
import java.time.LocalDateTime;

@MappedSuperclass
@Getter
@Setter
public class BaseEntity {

    private LocalDateTime createdDate;
    private LocalDateTime lastModifiedDate;

}

File: level-up-backend-gpt/level2-book-management-system/src/main/kotlin/org/example/level2bookmanagementsystem/jpa/entity/BaseEntity.kt
package org.example.level2bookmanagementsystem.jpa.entity

import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.ZonedDateTime


@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
class BaseEntity {

    @CreatedDate
    val createdAt: ZonedDateTime? = null

    @LastModifiedDate
    var updatedAt: ZonedDateTime = ZonedDateTime.now()

    var isDeleted: Boolean = false
}